### PR TITLE
feat(component): Implemented loading of styleguide react components

### PR DESCRIPTION
**Pull request by fragsalat**
I implemented the loading of the React components in the Aurelia app. This is the base to document the React components.
*Note*: I decided to mock the prop-types module with a simple file calles `src/prop-types.js`. I did this because the prop-types module is written for a nodejs environment using `process` variable and `module.exports` and therefore it is not usable with requirejs which is the default loader in an aurelia cli application.

The alias from react to preact happens via requirejs's map ability.
Beside that I just created a custom loader for Aurelia which wraps the React components in an Aurelia ViewModel.

### CODE REVIEW

File: src/app/feature/components/index.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import {React, render} from 'styleguide-web-components/imports';
+
+/**
+ * Configure the aurelia loader to use handle urls with !component
+ * @param config {FrameworkConfiguration}
+ */
+export function configure(config) {
+    const loader = config.aurelia.loader;
+    loader.addPlugin('styleguide-component', {
+        fetch(address) {
```
Comment (faboweb) [inline src/app/feature/components/index.js:12]: address is not defined?

File: src/app/feature/components/index.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+ * @param elementName {string}
+ */
+function wrapComponent(component, elementName) {
+    let bindableProps = [];
+    if (component.propTypes) {
+        bindableProps = Object.keys(component.propTypes).map(prop => bindable({
+            name: prop,
+            attribute: camelToKebab(prop),
+            changeHandler: 'render',
+            defaultBindingMode: 1
```
Comment (faboweb) [inline src/app/feature/components/index.js:58]: is there no enum for that? ^^

File: src/app/view/dynamic-html.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import {noView, ViewCompiler, ViewSlot, Container, ViewResources, bindable} from 'aurelia-framework';
+
+@noView()
+export class DynamicHtmlCustomElement {
```
Comment (faboweb) [inline src/app/view/dynamic-html.js:4]: what is this component for?

**Review by faboweb — APPROVED** [review]

**Comment by faboweb** [comment]
Hard to say much as I don't know much about aurelia

File: src/app/feature/components/index.js (hunk 2 above)
Comment (fragsalat) [reply src/app/feature/components/index.js:58]: Nope. I could create a constant for this.

File: src/app/feature/components/index.js (hunk 1 above)
Comment (fragsalat) [reply src/app/feature/components/index.js:12]: What do you mean with is not defined? It is a function parameter

File: src/app/view/dynamic-html.js (hunk 3 above)
Comment (fragsalat) [reply src/app/view/dynamic-html.js:4]: This component is used to render html dynamically with aurelia's template engine. The reason for this is that the React components are rendered via the aurelia template engine and loaded via the aurelia loader. So just rendering html would not render the React components. Therefore I introduced this custom element which takes the html to render and throw it into the template engine. The generated html is the final version we just append to the DOM.

**Pull request merged**